src/rules/__tests__/lowercase-name.test.ts

```diff
@@ -225,3 +225,37 @@ ruleTester.run('lowercase-name with ignore=it', rule, {
   ],
   invalid: [],
 });
+
+ruleTester.run('uppercase-name with ignore=top', rule, {
+  valid: [
+    {
+      code: `describe("Foo", () => {
+            test("bar", () => {
+             some_fn();
+           })
+        })`,
+      options: [{ ignore: [TopCase.top] }],
+    },
+    {
+      code: `describe("Foo", () => {
+                 it("bar", () => {
+                 some_fn();
+                })
+         })`,
+      options: [{ ignore: [TopCase.top] }],
+    },
+    {
+      code: 'describe(`Foo`, function () {})',
+      options: [{ ignore: [TopCase.top] }],
+    },
+    {
+      code: 'test(`Foo`, function () {})',
+      options: [{ ignore: [TopCase.top] }],
+    },
+    {
+      code: 'it(`Foo`, function () {})',
+      options: [{ ignore: [TopCase.top] }],
+    },
+  ],
+  invalid: [],
```

Comment:
Would you be able to add some `invalid` test cases as well please?

Would be great if you could include some multi-top level ones like in my above comment; and ideally mixing the casing of the inner functions 😄